db/hints/manager.cc

```diff
@@ -392,9 +392,6 @@ future<db::commitlog> manager::end_point_hints_manager::add_store() noexcept {
             cfg.fname_prefix = manager::FILENAME_PREFIX;
             cfg.extensions = &_shard_manager.local_db().extensions();
 
-            // HH doesn't utilize the flow that benefits from reusing segments.
-            // Therefore let's simply disable it to avoid any possible confusion.
-            cfg.reuse_segments = false;
```

Comment:
What happens during hints replay? Will the segments be converted into recycled segments instead of being deleted?

Reply:
Yes, up to the disk limit.